### DISCUSSION TITLE
Buffers don't free up memory / OS doesn't return freed up memory

### DISCUSSION THREAD

**Discussion by Br3ikros** [nodejs/help#4924] [Q&A]
### Version

16.13.1

### Platform

Windows, repl, heroku

### What steps will reproduce the bug?

simply run this:
```js
function getPrintFile() {

    let buffer = Buffer.alloc(1000000);

    return

}

function test() {

    for (let i = 0; i < 10; i++) {
        console.log(process.memoryUsage())
        getPrintFile()
    }


}

test()
console.log(process.memoryUsage())
```

### How often does it reproduce? Is there a required condition?

every time

### What is the expected behavior?

the external and rss memory not building up and actually releasing memory. this is a big issue when using a lot of buffers for example for image manipulation

### What do you see instead?

external and rss memory building up and not freeing

### Additional information

It doesn't even work if you assign to the buffer a null value after declaring it

**Comment by bnoordhuis** [comment]
I've converted this to a discussion because it's pretty much expected behavior, not a bug. Variations of what you see have been reported many times before, the keywords you're looking for are 'memory fragmentation' and 'garbage collector'.

**Reply by Br3ikros** [reply]
hi. ok. but the garbage collector doesn't work when using it. i installed jemalloc on my server, same thing. how is it not a bug or an issue if every single buffer doesn't free up the memory used?
